ion-schema/src/constraint.rs

```diff
@@ -204,26 +203,22 @@ impl Constraint {
     }
 
     /// Creates a [Constraint::Precision] from a [Range] specifying a precision range.
-    pub fn precision(precision: RangeImpl<usize>) -> Constraint {
-        Constraint::Precision(PrecisionConstraint::new(Range::NonNegativeInteger(
-            precision,
-        )))
+    pub fn precision(precision: U64Range) -> Constraint {
```

Comment:
It's not obvious, but here's why. All of the `*_length` constraints check a `len()` function that returns a `usize`. However, `Decimal::precision()` returns a `u64`. Rather than deal with a potentially lossy conversion if `usize` is not 64 bits, I'm just using both. I'll add a comment to that effect.

Now I'm curious to find out whether `u64` is the the best return type for `precision()`. (Maybe one of the non-zero types, even.)